FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m

```diff
@@ -0,0 +1,277 @@
+            @(FUIAccountSettingsOperationTypeUpdateName),
+        NSStringFromClass([FUIAccountSettingsOperationUpdatePassword class]) :
+            @(FUIAccountSettingsOperationTypeUpdatePassword),
+        NSStringFromClass([FUIAccountSettingsOperationForgotPassword class]) :
+            @(FUIAccountSettingsOperationTypeForgotPassword),
+        NSStringFromClass([FUIAccountSettingsOperationUpdateEmail class]) :
+            @(FUIAccountSettingsOperationTypeUpdateEmail),
+        NSStringFromClass([FUIAccountSettingsOperationUnlinkAccount class]) :
+            @(FUIAccountSettingsOperationTypeUnlinkAccount),
+        NSStringFromClass([FUIAccountSettingsOperationSignOut class]) :
+            @(FUIAccountSettingsOperationTypeSignOut),
+        NSStringFromClass([FUIAccountSettingsOperationDeleteAccount class]) :
+            @(FUIAccountSettingsOperationTypeDeleteAccount)
+      };
+    });
+  return [classMap[NSStringFromClass([self class])] integerValue];
+}
+
+#pragma mark - protected methods
+
+- (void)finishOperationWithError:(nullable NSError *)error {
+  if (error) {
+    switch (error.code) {
+      case FIRAuthErrorCodeEmailAlreadyInUse:
+        [self showAlertWithMessage:FUILocalizedString(kStr_EmailAlreadyInUseError)];
+        return;
+      case FIRAuthErrorCodeInvalidEmail:
+        [self showAlertWithMessage:FUILocalizedString(kStr_InvalidEmailError)];
+        return;
+      case FIRAuthErrorCodeWeakPassword:
+        [self showAlertWithMessage:FUILocalizedString(kStr_WeakPasswordError)];
+        return;
+      case FIRAuthErrorCodeTooManyRequests:
+        [self showAlertWithMessage:FUILocalizedString(kStr_SignUpTooManyTimesError)];
+        return;
+      case FIRAuthErrorCodeWrongPassword:
+        [self showAlertWithMessage:FUILocalizedString(kStr_WrongPasswordError)];
+        return;
+      case FIRAuthErrorCodeUserNotFound:
+        [self showAlertWithMessage:FUILocalizedString(kStr_UserNotFoundError)];
+        return;
+      case FIRAuthErrorCodeUserDisabled:
+        [self showAlertWithMessage:FUILocalizedString(kStr_AccountDisabledError)];
+        return;
+      case FUIAuthErrorCodeCantFindProvider:
+        {
+          NSString *message = [NSString stringWithFormat:FUILocalizedString(kStr_CantFindProvider),
+            error.userInfo[FUIAuthErrorUserInfoProviderIDKey]];
+          [self showAlertWithMessage:message];
+          return;
+        }
+      case FIRAuthErrorCodeUserMismatch:
+        [self showAlertWithMessage:FUILocalizedString(kStr_EmailsDontMatch)];
+        return;
+    }
+  }
+
+  [[FUIAuth defaultAuthUI] invokeOperationCallback:[self operationType] error:error];
+}
+
+- (void)showSelectProviderDialog:(nullable FUIAccountSettingsChooseProviderHandler)handler
+                      alertTitle:(nullable NSString *)title
+                    alertMessage:(nullable NSString *)message
+                alertCloseButton:(nullable NSString *)closeActionTitle {
+  UIAlertController *alert =
```

Comment:
FirebaseUI only supports 8+, so this is not an issue. We dropped iOS 7 when integrating Twitter auth.